assets/js/modules/analytics-4/components/dashboard/ActivationBanner/index.js

```diff
@@ -72,8 +117,33 @@ export default function ActivationBanner() {
 		}
 	}, [ hasEditScope, registry, returnToSetupStep, setValues ] );
 
-	const handleSubmit = useCallback( () => {
+	useEffect( () => {
+		if ( hasSetupBannerError ) {
+			setStep( ACTIVATION_STEP_REMINDER );
+		}
+	}, [ hasSetupBannerError ] );
+
+	const handleSubmit = useCallback( async () => {
 		if ( step === ACTIVATION_STEP_REMINDER ) {
+			// Clear errors before navigating to Setup Banner.
+			if ( hasSetupBannerError ) {
+				Object.keys( setupBannerErrors ).forEach(
+					async ( selector ) => {
+						const { store, args } =
+							setupBannerSelectors[ selector ];
+
+						await dispatch( store ).clearError(
+							selector,
+							args || []
+						);
+						await dispatch( store ).invalidateResolution(
+							selector,
+							args || []
+						);
```

Comment:
Can you wrap these two calls with `Promise.all()`?
```suggestion
						await Promise.all( [
							dispatch( store ).clearError(
								selector,
								args || []
							),
							dispatch( store ).invalidateResolution(
								selector,
								args || []
							),
						] );
```

Reply:
Updated, thank you!